db/view/view.cc

```diff
@@ -1232,6 +1235,74 @@ void view_updates::generate_update(
     }
 }
 
+std::optional<partition_key> view_updates::construct_view_partition_key_from_base(
+        const partition_key& base_pk)
```

Comment:
I'm considering to do a pre-computation in make_base_dependent_view_info of whether the view partition key is a permutation of base, so we could skip this function if it's not.